tests/cmd/git/commands.go

```diff
@@ -33,3 +39,36 @@ func Push(user model.User, keyPath string, app model.App, banner string) {
 	curlCmd = model.Cmd{CommandLineString: fmt.Sprintf(`curl -sL "%s"`, app.URL)}
 	Eventually(cmd.Retry(curlCmd, banner, cmdRetryTimeout)).Should(BeTrue())
 }
+
+// PushWithInterrupt executes a `git push deis master` from the current
+// directory using the provided key, but then halts the progress via SIGINT.
+func PushWithInterrupt(user model.User, keyPath string) {
+	sess := doGitPush(user, keyPath)
+	Eventually(sess.Err).Should(Say("Starting build... but first, coffee!"))
+
+	sess = sess.Interrupt()
+
+	newSess := doGitPush(user, keyPath)
+	Eventually(newSess.Err).ShouldNot(Say("exec request failed on channel 0"))
+	Eventually(newSess.Err).Should(Say("fatal: remote error: Another git push is ongoing"))
+	Eventually(newSess, settings.MaxEventuallyTimeout).Should(Exit(128))
+}
+
+// PushUntilResult executes a `git push deis master` from the current
+// directory using the provided key, until the command result satisfies
+// expectedCmdResult of type model.CmdResult, failing if
+// settings.DefaultEventuallyTimeout is reached first.
+func PushUntilResult(user model.User, keyPath string, expectedCmdResult model.CmdResult) {
+	envVars := append(os.Environ(), fmt.Sprintf("DEIS_PROFILE=%s", user.Username))
+	pushCmd := model.Cmd{Env: envVars, CommandLineString: fmt.Sprintf(
+		pushCommandLineString, settings.GitSSH, keyPath)}
+
+	Eventually(cmd.RetryUntilResult(pushCmd, expectedCmdResult, 5*time.Second,
+		settings.DefaultEventuallyTimeout)).Should(BeTrue())
```

Comment:
@arschles yes, it appears it and `MaxDefaultEventuallyTimeout` can be overridden via env vars: https://github.com/deis/workflow-e2e/blob/master/tests/settings/settings.go#L20